### PR TITLE
Revisit value handling in the query builder

### PR DESCRIPTION
**Pull request by olim7t**
- only allow QueryOuterClass.Value as literals
- allow explicit bind markers

### CODE REVIEW

File: sgv2-service-common/src/main/java/io/stargate/sgv2/common/cql/builder/Literal.java [hunk 1]
```diff
@@ -15,15 +15,20 @@
  */
 package io.stargate.sgv2.common.cql.builder;
 
-class Literal implements Term {
+import io.stargate.bridge.proto.QueryOuterClass.Value;
+import java.util.Objects;
 
-  private final Object value;
+public class Literal implements Term {
```
Comment (ivansenic) [inline sgv2-service-common/src/main/java/io/stargate/sgv2/common/cql/builder/Literal.java:21]: mentioned already in #1796 , this is a nice candidate for immutable..

**Review by ivansenic — APPROVED** [review]
Seems good to me.. Hope that tests would catch anything that I might have missed..

File: sgv2-service-common/src/main/java/io/stargate/sgv2/common/cql/builder/Literal.java (hunk 1 above)
Comment (olim7t) [reply sgv2-service-common/src/main/java/io/stargate/sgv2/common/cql/builder/Literal.java:21]: Maybe we can address this in a follow-up ticket, I don't want to delay this...

But I think we are over-using `@Value.Immutable`: in this case the class is already immutable and it doesn't need value equality. So it wouldn't add any value.

Comment (olim7t) [reply sgv2-service-common/src/main/java/io/stargate/sgv2/common/cql/builder/Literal.java:21]: And just after writing this, I notice that I've implemented `equals` and `hashCode` on another branch... But still, it's not that hard to do manually.

**Pull request merged**